internal/rest/resources/control.go

```diff
@@ -126,6 +134,30 @@ func controlPost(state state.State, r *http.Request) response.Response {
 		}
 	})
 
+	// Replace the server keypair if the cluster member name has changed upon initialization.
+	if !certNameMatches {
+		err := os.Remove(filepath.Join(state.FileSystem().StateDir, "server.crt"))
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		err = os.Remove(filepath.Join(state.FileSystem().StateDir, "server.key"))
+		if err != nil {
+			return response.SmartError(err)
+		}
```

Comment:
Do we need a reverter that resets and reloads the old certificates?

Reply:
I don't think so, we are arbitrarily replacing it here which means it should not be relied upon to remain static prior to init. Once we reset the daemon, we will generate a new keypair again.

Reply:
Ok I see.